n3fit/src/n3fit/backends/keras_backend/MetaModel.py

```diff
@@ -411,51 +410,78 @@ def load_identical_replicas(self, model_file):
         """
         From a single replica model, load the same weights into all replicas.
         """
-        weights = self._format_weights_from_file(model_file)
+        single_replica = self.single_replica_generator()
+        single_replica.load_weights(model_file)
+        weights = single_replica.get_replica_weights(0)
 
         for i_replica in range(self.num_replicas):
             self.set_replica_weights(weights, i_replica)
 
-    def _format_weights_from_file(self, model_file):
-        """Read weights from a .h5 file and format into a dictionary of tf.Variables"""
-        weights = {}
 
-        with h5py.File(model_file, 'r') as f:
-            # look at layers of the form NN_i and take the lowest i
-            i_replica = 0
-            while f"{NN_PREFIX}_{i_replica}" not in f:
-                i_replica += 1
+def stacked_single_replicas(layer):
+    """
+    Check if the layer consists of stacked single replicas (Only happens for NN layers)
```

Comment:
could you add a bit more to this docstr about why this is needed?

Maybe since it is a check, a better function name would be `is_stacked_single_replicas`?

Reply:
In the context of this PR, for the preprocessing layer we extract replica weights by taking the right slice of the multi-replica preprocessing layer, while the NN (dense) layers are not joined yet and need to be extracted in a different way. So here this could be replaced with simply a check if the layer is a preprocessing or a nn layer.
However in the next PR, NNs will also be joined together into one layer, so that won't be possible, and still we need to preserve the old way of extracting weights just for the dense_per_flavour layer, where replicas are not joined. (I guess it's still possible to do there, but I didn't want to waste time on that. I have added `is_` and explained a little, is that ok?

Reply:
Yes. My comment was mostly about giving it some context.